src/modules-by-example.adoc

```diff
@@ -0,0 +1,321 @@
+This clause expects a list of strings, and both the syntax and semantics are the same as the
+`symbols` field of `$ion_shared_symbol_table`.
+
+Once this module is defined, we can _install_ it into the local symbol table:
+
+[{nrm}]
+----
+  install_symbols: [ local ]
+----
+
+This field expects a list of symbols that match names declared in the `modules` field.  The
+resulting local symbol table is simply the concatenation of the exported symbol tables of those
+modules.  This works the same way as the `imports` field of `$ion_symbol_table`.
+
+
+=== Importing Symbols
+
+Given the equivalencies above, we could perform a naive round-trip of the preceding 1.1 document
+back to 1.0. First, turn the `local` module into the equivalent shared symbol table:
+
+[{nrm}]
+----
+**$ion_shared_symbol_table**::{
+  name: "extracted",
+  version: 1,
+  symbols: ["s1", "s2"]
+}
+----
+
+Then translate `install_symbols:[local]` into its 1.0 equivalent:
+
+[{nrm}]
+----
+*$ion_1_0*
+**$ion_symbol_table**::{
+  imports: [{ name: "extracted", version: 1, max_id: 2 }]
+}
+----
+
+NOTE:  Even ignoring Ion 1.1, this is how you would extract local symbols into a
+new shared symbol table.
+
+The latter imports-only document has this 1.1 equivalent:
+
+
+[{nrm}]
+----
+*$ion_1_1*
+**$ion_encoding**::{
+  modules: [
+    (*import* local "extracted" 1 2)
+  ],
+  install_symbols: [ local ]
+}
+----
+
+Here we see a new form inside the `modules` field that imports a module into the encoding
+environment and assigns it a name.
+The `*import*` keyword starts an S-expression that expects three or four arguments. The first is
+a symbolic name that we can use later to refer to the imported module.  The remaining arguments
+are effectively the `name`, `version` and `max_id` fields of the 1.0 `imports` struct, with only
+the max_id being optional in this form.
+
+TIP: From the perspective of Ion 1.1, shared symbol tables _are_ encoding modules.
+
+
+=== Declaring Multiple Modules
+
+Let’s look at a scenario with both imported and locally-defined symbols:
+
+[{nrm}]
+----
+*$ion_1_0*
+**$ion_symbol_table**::{
+  imports: [{ name: "shared1", version: 1, max_id: 10 },
+            { name: "shared2", version: 2, max_id: 20 }],
+  symbols: ["s1", "s2"]
+}
+----
+
+Here’s the Ion 1.1 equivalent in terms of symbol allocation order:
+
+[{nrm}]
+----
+*$ion_1_1*
+**$ion_encoding**::{
+  modules: [(*import* n1 "shared1" 1 10),
+            (*import* n2 "shared2" 2 20),
+            (*module* local (*symbols* ["s1", "s2"]))],
+  install_symbols: [n1, n2, local]
+}
+----
+
+Just as in the 1.0 version, this allocates ten symbol IDs for `shared1` (as requested by its
+max_id argument), twenty symbol IDs for `shared2`, then the two locally-defined symbols.
+
+By decoupling symbol-table importing from installation, Ion 1.1 allows some encoding techniques
+that are not possible in 1.0.  For example, we can give local symbols smaller IDs than imported
+symbols by installing `local` first:
```

Comment:
Consider adding a sentence after the example explaining why smaller symbol IDs are beneficial.